components/log/log.h

```diff
@@ -0,0 +1,102 @@
+#ifndef DATADOG_PHP_PROFILER_LOG_H
+#define DATADOG_PHP_PROFILER_LOG_H
+
+#include <components/string_view/string_view.h>
+#include <pthread.h>
+#include <stdint.h>
+
+#if __cplusplus
+#define C_STATIC(...)
+#else
+#define C_STATIC(...) static __VA_ARGS__
+#endif
+
+typedef enum {
+    DATADOG_PHP_LOG_UNKNOWN = -1,
+    DATADOG_PHP_LOG_OFF = 0,
+    DATADOG_PHP_LOG_ERROR,
+    DATADOG_PHP_LOG_WARN,
+    DATADOG_PHP_LOG_INFO,
+    DATADOG_PHP_LOG_DEBUG,
+} datadog_php_log_level;
```

Comment:
Nit for later: @krakjoe brought up a good point that having `php` in the name makes this seem like a Zend Engine component. I believe the original reason for adding `php` in the namespace was to disambiguate it from other languages within APM, but that is assuming that this would become part of a shared library that other languages would use. I don't believe that will be the case so because the semantics of `php` for extensions in the wild means "Zend Engine", the name is a bit confusing.

This isn't a blocker, but after all the components are merged, we might want to discuss a naming scheme for components.

Reply:
> I believe the original reason for adding php in the namespace was to disambiguate it from other languages within APM, but that is assuming that this would become part of a shared library that other languages would use.

Sort of: I am more worried about us having a symbol that conflicts with something in a datadog shared library, like `libddprof` or a theoretical `libdatadog` or `libddapm` or something which might also share the `datadog_php_` prefix. We could probably shorten it to `dd_php_` or something, but I want to avoid `trace` if possible because we're moving beyond just trace products.